test/lib/validators/content_view_puppet_module_validator_test.rb

```diff
@@ -5,52 +5,83 @@
       @validator.validate(@model)
 
       refute_empty @model.errors[:base]
     end
 
-    test "fails if only name provided" do
-      @model = OpenStruct.new(@base_record.merge(:name => "module name"))
+    test "fails if only author provided" do
+      @model = OpenStruct.new(@base_record.merge(author: "johndoe"))
       @validator.validate(@model)
 
       refute_empty @model.errors[:base]
     end
 
-    test "passes if name and author provided" do
-      puppet = PuppetModule.create!(:name => "module name", :author => "module author", :uuid => '9932943299423')
-      puppet.repositories << @repository
-      @model = OpenStruct.new(@base_record.merge(:name => "module name", :author => "module author"))
+    test "fails if name and author do not match a puppet module" do
+      # fixture katello_puppet_modules(:abrt)
+      @model = OpenStruct.new(@base_record.merge(name: "abrt", author: "janedoe"))
       @validator.validate(@model)
-      assert_empty @model.errors[:base]
+      refute_empty @model.errors[:base]
+    end
+
+    test "fails if both name and uuid blank" do
+      @model = OpenStruct.new(errors: {base: []})
+      @validator.validate(@model)
+
+      refute_empty @model.errors[:base]
     end
 
-    test "passes if uuid provided" do
-      @model = OpenStruct.new(@base_record.merge(:uuid => "3bd47a52-0847-42b5-90ff-206307b48b22"))
+    test "passes if uuid matches a puppet module" do
+      # fixture katello_puppet_modules(:abrt)
+      @model = OpenStruct.new(@base_record.merge(uuid: "3bd47a52-0847-42b5-90ff-20630asddfat"))
       @validator.validate(@model)
 
       assert_empty @model.errors[:base]
     end
 
-    test "fails if module does not exists" do
-      @model = OpenStruct.new(@base_record.merge(:name => "module name", :author => "module author"))
+    test "fails if uuid does not match a puppet module" do
+      # fixture katello_puppet_modules(:abrt)
+      @model = OpenStruct.new(@base_record.merge(uuid: "3bd47a52-90ff-20630asddfat"))
       Katello::PuppetModule.stubs(:exists?).returns(false)
       @validator.validate(@model)
 
       refute_empty @model.errors[:base]
     end
 
-    test "passes if the module does exist" do
-      puppet = PuppetModule.create!(:name => "module name", :author => "module author", :uuid => '9932943299423')
-      puppet.repositories << @repository
-      @model = OpenStruct.new(@base_record.merge(:name => "module name", :author => "module author"))
+    test "fails if puppet module does not belong to a repository in the content view organization" do
+      # fixture katello_puppet_modules(:abrt) - Empty Organization
+      @model = OpenStruct.new(@base_record.merge(uuid: "1234", content_view: OpenStruct.new(organization: katello_environments(:dev))))
+      @validator.validate(@model)
+
+      refute_empty @model.errors[:base]
+    end
+
+    test "passes if identical puppet modules belong to repositories of different organizations" do
```

Comment:
Thanks! How about the failure case as that is one we want to ensure also is prevented?